test/unit/grammar/range_rule.cpp

```diff
@@ -10,23 +10,57 @@
 // Test that header file is self-contained.
 #include <boost/url/grammar/range_rule.hpp>
 
+#include <boost/url/grammar/alpha_chars.hpp>
 #include <boost/url/grammar/char_rule.hpp>
 #include <boost/url/grammar/parse.hpp>
+#include <boost/url/grammar/sequence_rule.hpp>
+#include <boost/url/grammar/token_rule.hpp>
 
 #include "test_suite.hpp"
 
+#include <algorithm>
+#include <initializer_list>
+
 namespace boost {
 namespace urls {
 namespace grammar {
 
 struct range_rule_test
 {
+    template<class R>
+    static
+    void
+    check(
+        string_view s,
+        std::initializer_list<
+            string_view> init,
+        R const& r)
+    {
+        auto rv = parse(s, r);
+        if(! BOOST_TEST(rv.has_value()))
+            return;
+        if(! BOOST_TEST_EQ(
+                rv->size(), init.size()))
+            return;
+        BOOST_TEST(
+            std::equal(
+                rv->begin(),
+                rv->end(),
+                init.begin()));
+    }
+
```

Comment:
You may want to refactor this to use instead use:
```cpp
BOOST_TEST_ALL_EQ(rv->begin(), rv->end(), init.begin(), init.end());
```
which gives some slightly better error diagnostics on failure.

Reply:
I don't think I have that, this isn't Lightweight Test